p4-16/psa/PSA.mdk

```diff
@@ -382,16 +382,50 @@ recirculated, followed by the payload of that packet, i.e. the part
 that was not parsed by the egress parser.  Truncation is not supported
 for RECIRC packets.
 
-### User-defined metadata for all ingress packets
+### User-defined metadata for all ingress packets {#sec-ingress_initial-metadata}
 
 The PSA architecture does not mandate initialization of user-defined
 metadata to known values as given as input to the ingress parser.  If
 a user's P4 program explicitly initializes all user-defined metadata
 early on (e.g. in the parser's `start` state), then that will flow
-through the rest of the parser into the `Ingress`  control block as one
+through the rest of the parser into the `Ingress` control block as one
 might normally expect.  This will be left as an option to the user in
 their P4 programs, not required behavior for all P4 programs.
 
+There are two direction `in` parameters to the ingress parser with
+user-defined types, named `resubmit_meta` and `recirculate_meta`.
+They may be used to carry metadata for resubmitted and recirculated
+packets.
+
+Consider a packet that arrives at the ingress pipeline, and during
+ingress processing the P4 program assigns values to fields of the PSA
+standard metadata such that the packet is resubmitted (see Section
+[#sec-after-ingress] for details on how to do so).  In the ingress
+deparser, the P4 program assigns a value to the `out` parameter named
+`resubmit_meta`.  This value (which can be a collection of many
+individual values in fields, sub-structs, headers, etc.) becomes
+associated with the resubmitted packet by the PSA implementation, and
+when the resubmitted packet begins ingress parsing, that becomes the
+value of the `in` parameter named `resubmit_meta` to the ingress
+parser.
+
+For resubmitted packets, the value of the `in` parameter named
+`recirculate_meta` is undefined.
+
+Conversely, for recirculated packets, the value of the `in` parameter
+named `recirculate_meta` contains whatever value was assigned to the
+egress deparser `out` parameter named `recirculate_meta` when the
+packet was recirculated.  The value of the `in` parameter
+`resubmit_meta` is undefined for recirculated packets.
+
+For packets from a port, including the CPU port, both of the `in`
+parameters `resubmit_meta` and `recirculate_meta` have undefined
+contents.
+
+A P4 program that wishes to maximize portability must not use an
```

Comment:
I can't find such a global statement.  I'm happy to add one, and not mention it here, if you prefer.  Let me know.

Reply:
I looked as well and didn't see one either, but this is something that keeps coming back. Let's add one.

Reply:
I have moved that statement to the end of section 1, with another general guideline.  I'll let you review it before anything is merged in, in case it is awkwardly worded or placed.